apps/pkeyutl.c

```diff
@@ -277,25 +289,8 @@ int pkeyutl_main(int argc, char **argv)
     if (!app_RAND_load())
         goto end;
 
-    if (rawin && pkey_op != EVP_PKEY_OP_SIGN && pkey_op != EVP_PKEY_OP_VERIFY) {
-        BIO_printf(bio_err,
-                   "%s: -rawin can only be used with -sign or -verify\n",
-                   prog);
-        goto opthelp;
-    }
-
-    if (digestname != NULL && !rawin) {
-        BIO_printf(bio_err,
-                   "%s: -digest can only be used with -rawin\n",
-                   prog);
-        goto opthelp;
-    }
-
-    if (rawin && rev) {
-        BIO_printf(bio_err, "%s: -rev cannot be used with raw input\n",
-                   prog);
-        goto opthelp;
-    }
+    if (digestname != NULL)
+        rawin = 1;
```

Comment:
This is wrong, because when the input **is** a digested value, one might (especially with RSA PKCS#1) want to indicate which digest the input was constructed with. The `-digest <...>` option is in fact especially useful with `rawin` false, with `rawin` true, the default digest is generally a sensible choice, though with RSA there are often multiple valid choices (RSA with SHA2-256 and RSA with SHA2-512).

Reply:
You're mixing up `-digest {name}` with `-pkeyopt digest:{name}`.  The latter is used when the input is a digested value (and it usually has a default too).

Reply:
Thanks for the clue. Good to know.